src/current.rs

```diff
@@ -32,3 +32,19 @@ pub fn clock() -> VectorClock {
 pub fn clock_for(task_id: TaskId) -> VectorClock {
     ExecutionState::with(|state| state.get_clock(task_id).clone())
 }
+
+/// Sets the `tag` field of the current task.
+pub fn set_tag_for_current_task(tag: Tag) {
+    ExecutionState::set_tag_for_current_task(tag);
+}
+
+/// Gets the `tag` field of the current task.
+pub fn get_tag_for_current_task() -> Tag {
+    ExecutionState::get_tag_for_current_task()
+}
+
+/// Gets the `TaskId` of the current task
+/// NOTE: Will panic if there is no current task.
+pub fn get_current_task() -> TaskId {
```

Comment:
Could we change this to return an `Option<TaskId>` and use `ExecutionState::try_current`, so that this can be called from a context where there may be no current task?

Reply:
I was thinking we might want to use it in a custom scheduler (which does not have access to internal types, but is called from within Shuttle).